include/kimera_vio_ros/RosDataProviderInterface.hpp

```diff
@@ -159,30 +167,31 @@ class RosDataProviderInterface : public DataProviderInterface {
 
  private:
   // Define publisher for debug images.
-  image_transport::Publisher debug_img_pub_;
+  // image_transport::Publisher debug_img_pub_;
+
+  // Typedefs
+//  typedef pcl::PointCloud<pcl::PointXYZRGB> PointCloudXYZRGB;
 
   // Publishers
-  ros::Publisher pointcloud_pub_;
+//  rclcpp::Publisher<PointCloudXYZRGB> pointcloud_pub_;
   // Published 3d mesh per frame (not time horizon of opitimization!)
-  ros::Publisher mesh_3d_frame_pub_;
-  ros::Publisher odometry_pub_;
-  ros::Publisher resiliency_pub_;
-  ros::Publisher frontend_stats_pub_;
-  ros::Publisher imu_bias_pub_;
-  ros::Publisher trajectory_pub_;
-  ros::Publisher posegraph_pub_;
+  rclcpp::Publisher<pcl_msgs::msg::PolygonMesh>::SharedPtr mesh_3d_frame_pub_;
+  rclcpp::Publisher<nav_msgs::msg::Odometry>::SharedPtr odometry_pub_;
+  rclcpp::Publisher<std_msgs::msg::Float64MultiArray>::SharedPtr resiliency_pub_;
+  rclcpp::Publisher<std_msgs::msg::Float64MultiArray>::SharedPtr frontend_stats_pub_;
+  rclcpp::Publisher<std_msgs::msg::Float64MultiArray>::SharedPtr imu_bias_pub_;
+  rclcpp::Publisher<nav_msgs::msg::Path>::SharedPtr trajectory_pub_;
+  rclcpp::Publisher<pose_graph_msgs::msg::PoseGraph>::SharedPtr posegraph_pub_;
+  rclcpp::Publisher<sensor_msgs::msg::Image>::SharedPtr debug_img_pub_;
 
   // Define tf broadcaster for world to base_link (IMU) and to map (PGO).
-  tf::TransformBroadcaster tf_broadcaster_;
+  tf2_ros::TransformBroadcaster tf_broadcaster_;
```

Comment:
Can't we keep compatibility between ROS and ROS2?
Or is that not possible?

Reply:
The C++ client library for ROS2 is rclcpp, and breaks API compatibility with ROS1 roscpp. Also, tf1 was deprecated around ROS1 Indigo arround ~2015 for the newer tf2 library, which is the only tf library version supported by ROS2.

Reply:
Agree with tf1 vs tf2 (I don't know why tf1 is there :sweat_smile:).